great_expectations/profiler/profiler_rule/rule_state.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Dict, List, Optional, Union
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core import IDDict
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.profiler.parameter_builder.parameter import Parameter
+
+DOMAIN_KWARGS_PARAMETER_NAME: str = "domain_kwargs"
+DOMAIN_KWARGS_PARAMETER_FULLY_QUALIFIED_NAME: str = (
+    f"$domain.{DOMAIN_KWARGS_PARAMETER_NAME}"
+)
+VARIABLES_KEY: str = "$variables."
+
+
+# TODO: <Alex>ALEX -- We should consider changing the names "parameters" and "Parameter.parameters" to be more semantically meaningful.</Alex>
```

Comment:
💯

Reply:
@talagluck I will try this in the next commit.

Reply:
Not done yet.

Reply:
Actually, @talagluck and @anthonyburdi -- we should discuss this when we get a chance -- to make sure that we choose the best alternative for ourselves.  (Mentioning @gilpasternak35 -- as an FYI -- but no actions requested, since you have midterms now.). Thanks!

Reply:
This is improved upon in PR #2718 /cc @talagluck @anthonyburdi